libraries/tuttle/src/tuttle/host/memory/MemoryPool.cpp

```diff
@@ -166,17 +178,44 @@ boost::intrusive_ptr<IPoolData> MemoryPool::allocate( const std::size_t size )
 
 	if( pData != NULL )
 	{
-		pData->_size = size;
```

Comment:
We need to update the "size" used (useful to compute the wasted size for example). The object still contains the real allocated size, called "reservedSize".